src/light.js

```diff
@@ -20,20 +24,26 @@ class gltfLight
         fromKeys(this, jsonLight);
     }
 
-    toUniform(transform)
+    toUniform(gltf)
     {
         let uLight = new UniformLight();
-        let rotation = mat3.create();
-        mat4.getRotation(rotation, transform);
-        mat3.multiply(uLight.direction, uLight.direction, rotation);
+
+        if (this.node !== undefined)
+        {
+            let transform = gltf.nodes[this.node].worldTransform;
+            let rotation = quat.create();
+            let alongNegativeZ = vec3.fromValues(0, 0, -1);
+            mat4.getRotation(rotation, transform);
+            vec3.transformQuat(uLight.direction, alongNegativeZ, rotation);
+            mat4.getTranslation(uLight.position, transform);
+        }
 
         uLight.range = this.range;
```

Comment:
Even though `UniformLight` correctly initializes `range` to -1, it will be overwritten by `this.range` which is `undefined` after the constructor when it isn't given. So we'll be uploading `undefined` to the uniform, and IDK what will happen. So the trick is to default initialize it in the constructor, and when `fromJson` is called, it will automatically determine if it should populate `range` with valid values or if it should keep the default ones.